### DISCUSSION TITLE
FORBIDDEN access to worker's logs

### DISCUSSION THREAD

**Discussion by denisvolokh** [apache/airflow#17251] [Q&A] [answered]
**Apache Airflow version:**
Airflow version: 2.1.2

Airflow Web UI is not able to fetch logs (actually stopped working) from the worker's container. It fails with the following error:

> *** Log file does not exist: /app/airflow/logs/PROD-synch_strategies_data/t_synch_strategies_summaries/2021-07-27T06:08:44.532400+00:00/1.log
*** Fetching from: http://a32f9e563f01:8793/log/PROD-synch_strategies_data/t_synch_strategies_summaries/2021-07-27T06:08:44.532400+00:00/1.log
*** Failed to fetch log file from worker. 403 Client Error: FORBIDDEN for url: http://a32f9e563f01:8793/log/PROD-synch_strategies_data/t_synch_strategies_summaries/2021-07-27T06:08:44.532400+00:00/1.log
For more information check: https://httpstatuses.com/403

When I checked the logs at the worker container, I can see all logs at the worker's container.

What could be wrong here?

**Comment by potiuk** [comment, marked as answer]
You need to have Webserver secret key the same in webserver and workers: https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#secret-key

**Reply by denisvolokh** [reply]
Thank you, secret key helped to resolve my problem

**Reply by mujohiddin** [reply]
@denisvolokh Can you tell me how you fixed it?

**Reply by piyush168713** [reply]
Ref: https://airflow.apache.org/docs/helm-chart/stable/production-guide.html#webserver-secret-key

**Comment by mnaranjorion** [comment]
Good morning,
I am installing an Airflow cluster with CeleryExecutor. The deployment of the services on the different machines works correctly, and the distribution of tasks when executing a DAG apparently also. 
However, I can't see the logs of the tasks executed in the nodes where the WebServer is not present, although the secret_key of the 'airflow.cfg' file is the same for all the nodes. 
I have also checked the time and set the same timezone for all machines. 

Same error as in the first message from @denisvolokh.
```bash
*** Log file does not exist: <path_to_logs>/logs/dag_id=example_branch_labels/run_id=manual__2022-05-09T09:05:11.152836+00:00/task_id=task1/attempt=1.log
*** Fetching from: http://<server_name>:8793/log/dag_id=example_branch_labels/run_id=manual__2022-05-09T09:05:11.152836+00:00/task_id=task1/attempt=1.log
*** !!!! Please make sure that all your Airflow components (e.g. schedulers, webservers and workers) have the same 'secret_key' configured in 'webserver' section and time is synchronized on all your machines (for example with ntpd) !!!!!
****** See more at https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#secret-key
****** Failed to fetch log file from worker. Client error '403 FORBIDDEN' for url 'http://<server_name>:8793/log/dag_id=example_branch_labels/run_id=manual__2022-05-09T09:05:11.152836+00:00/task_id=task1/attempt=1.log'
For more information check: https://httpstatuses.com/403
```

What configuration could I be missing?

thank you very much,

**Reply by potiuk** [reply]
From https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html

> Use the same configuration across all the Airflow components. While each component does not require all, some configurations need to be same otherwise they would not work as expected. A good example for that is [secret_key](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#config-webserver-secret-key) which should be same on the Webserver and Worker to allow Webserver to fetch logs from Worker.

**Reply by mnaranjorion** [reply]
Thank you very much for your reply,

as I indicated, the value of the `secret_key` field in the `airflow.cfg` file on all the workers' machines is the same. That's my main problem, I can't find any extra configuration in this regard. What I have tried so far:

* Same `secret_key` in all `airflow.cfg` configuration files on all machines.
* Set timezone so that all machines have the same time.
* Restart `scheduler`, `web server` and `worker` services after changes.

Is there anything else I should be aware of?

Thanks

**Reply by potiuk** [reply]
Seting TIME not timezone on all machines is important. Make sure time is synchronized on all your machines. Timezone does not matter, but syncing time (with ntp for example) is crucial.

**Reply by potiuk** [reply]
Also -you might have some firewalls in between your webserver and workers - so you need to track down any networking you have between webserver and worker.

**Reply by potiuk** [reply]
Another thing is - maybe you somewhat override the secret. There are many ways airflow configuration variables are set - https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html. - variables, secrets, _CMD. Maybe you have different settings outside of teh .cfg - or maybe you are using different .cfg file on one of your webserver/worker altogether.

I saw that happening in the past.

**Reply by mnaranjorion** [reply]
Good morning,
After fixing a configuration problem with NTP, I have managed to view the log files of workers installed on machines other than the WebServer.

Thank you very much for your help.

**Reply by mbhavik91** [reply]
@mnaranjorion : Could you please share details on how you fixed configuration problem with NTP?

**Comment by manojnkhunte** [comment]
firstly login to all airflow containers like scheduler,webserver,worker and see if you see same time . if yes it means time is already synchronised. 
then place below variable AIRFLOW__WEBSERVER__SECRET_KEY like below .... 
x-airflow-common:
  &airflow-common
  image: ${AIRFLOW_IMAGE_NAME:-apache/airflow:2.10.0}
  # build: .
  environment:
    &airflow-common-env
    AIRFLOW__CORE__EXECUTOR: CeleryExecutor
    AIRFLOW__DATABASE__SQL_ALCHEMY_CONN: postgresql+psycopg2://airflow:airflow@postgres/airflow
    AIRFLOW__CELERY__RESULT_BACKEND: db+postgresql://airflow:airflow@postgres/airflow
    AIRFLOW__CELERY__BROKER_URL: redis://:@redis:6379/0
    AIRFLOW__CORE__FERNET_KEY: ''
    **AIRFLOW__WEBSERVER__SECRET_KEY: 'your_airflow_webserver_sec_key'**
you can copy paste AIRFLOW__WEBSERVER__SECRET_KEY as it is .
.... it works with this , hope it helps

**Comment by piyush168713** [comment]
Ref: https://airflow.apache.org/docs/helm-chart/stable/production-guide.html#webserver-secret-key
